app/assets/javascripts/discourse/app/services/lightbox.js

```diff
@@ -0,0 +1,266 @@
+import {
+  DOCUMENT_ELEMENT_LIGHTBOX_OPEN_CLASS,
+  LIGHTBOX_APP_EVENT_NAMES,
+  MIN_CAROUSEL_ARROW_ITEM_COUNT,
+  SELECTORS,
+} from "discourse/lib/lightbox/constants";
+import Service, { inject as service } from "@ember/service";
+import {
+  getSiteThemeColor,
+  setSiteThemeColor,
+} from "discourse/lib/lightbox/helpers";
+
+import Mobile from "discourse/lib/mobile";
+import { bind } from "discourse-common/utils/decorators";
+import { isDocumentRTL } from "discourse/lib/text-direction";
+import { processHTML } from "discourse/lib/lightbox/process-html";
+
+export default class LightboxService extends Service {
+  @service appEvents;
+
+  lightboxIsOpen = false;
+  lightboxClickElements = [];
+  lastFocusedElement = null;
+  originalSiteThemeColor = null;
+  onFocus = null;
+  selector = null;
+
+  callbacks = {};
+  options = {};
+
+  async init() {
+    super.init(...arguments);
+
+    this.callbacks = {
+      onOpen: this.onLightboxOpened,
+      onClose: this.onLightboxClosed,
+      onWillChange: this.onLightboxItemWillChange,
+      onItemDidChange: this.onLightboxItemDidChange,
+      onCleanUp: this.onLightboxCleanedUp,
+    };
+
+    this.options = {
+      isMobile: Mobile.mobileView,
```

Comment:
why does it need to be an option? We should also just use the service. (@service site; this.site.mobileView)